tests/rptest/tests/cloud_storage_inventory_test.py

```diff
@@ -0,0 +1,121 @@
+import base64
+from pathlib import Path
+
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import SISettings
+from rptest.tests.prealloc_nodes import PreallocNodesTest
+from rptest.util import Scale
+from rptest.utils.si_utils import NTP
+
+
+class CloudStorageInventoryTest(PreallocNodesTest):
+    def __init__(self, test_context):
+        self.test_context = test_context
+        super().__init__(test_context=self.test_context,
+                         node_prealloc_count=1,
+                         num_brokers=3,
+                         si_settings=SISettings(test_context=test_context))
+        self.scale = Scale(self.test_context)
+        self.topics = (
+            TopicSpec(name='topic-A', partition_count=3, replication_factor=3),
+            TopicSpec(name='topic-B', partition_count=3, replication_factor=3),
+        )
+        self.extra_rp_conf = {
+            'cloud_storage_inventory_based_scrub_enabled': True,
+            'cloud_storage_enable_scrubbing': True,
+            'cloud_storage_inventory_self_managed_report_config': True,
+            'cloud_storage_inventory_report_check_interval_ms': 10000
+        }
+
+        self.redpanda.set_extra_rp_conf(self.extra_rp_conf)
+
+        # A gz CSV which contains the following rows among others for testing
+        """
+        "a-bucket","5ad6cd22/kafka/topic-A/110_24/626-631-1573504-1-v1.log.2"
+        "a-bucket","8bb8847f/kafka/topic-A/0_24/494-499-1573504-1-v1.log.2"
+        "a-bucket","bf3cd6b8/kafka/topic-B/0_24/464-469-1573504-1-v1.log.2"
+        "a-bucket","d15817a9/kafka/topic-B/2_24/560-565-1573504-1-v1.log.2"
```

Comment:
Just noting that for newer clusters, the path structure will look a bit different. Do we also look for manifests when we consume the inventory file?

It seems like the same path expr should work though
https://github.com/redpanda-data/redpanda/blob/5c5b4ef8ae6f8c8565d196d2ba2d7fd725ef346e/src/v/cloud_storage/segment_path_utils.h#L32

Reply:
We create hashes for the manifest files but right now we do not use them while scrubbing, because the manifests need to be downloaded right after a HEAD call. See https://github.com/redpanda-data/redpanda/pull/22810/commits/661d102b6e748e3c662c9593082aa085013c40e4 for how the hashes are planned to be used, right now we only check for segment paths. There is no reason why we couldn't eventually do this for manifests too, though.

Would there need to be some changes to read the manifest paths from the report? Ultimately the report contains paths to keys in bucket, and the scrubber should also check for the exact same paths (since that's the only way to access the object) so the core logic should remain unchanged, but maybe the tests and expressions used to validate paths would need to be adjusted?

Reply:
>Would there need to be some changes to read the manifest paths from the report? 

Not right now, given we're only validating data segments here.

>the core logic should remain unchanged, but maybe the tests and expressions used to validate paths would need to be adjusted?

Agreed, I don't think there's much of a change in how this works. Just wanted to call out that if any regexes or tests do exist that depend on the previous hashing scheme, they'll need to be updated moving forward

Reply:
BTW, I wonder how difficult it'd be to write a python utility to generate an inventory file using the s3 client list operation. Then we could get a flavor of testing real-ish scenarios in ducktape.

Not a suggestion for this PR, but something we should probably consider as the scrubber becomes more functional

Reply:
Yeah, I plan to add more tests in the ducktape suite added in this PR, for some of the newer tests we'll need non-static data so we will have to create a report from the bucket.